### PR TITLE
executor: fix cop request key range order another case

### LINKED ISSUE #36876: tidb oom because of infinite loop
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->
pk_snpre10114 only have 103 rows, but TableRangeScan operator actRows is 16002212
![image](https://user-images.githubusercontent.com/5506767/182751830-26fea09c-5d34-43ef-8c11-6f9783c3d8b2.png)
![image](https://user-images.githubusercontent.com/5506767/182751883-3968d44a-6c11-4ad4-8e0f-a35006f1ae1c.png)
```
MySQL [coms]> select count(*) from pk_snpre10114;
+----------+
| count(*) |
+----------+
|      103 |
+----------+
1 row in set (0.00 sec)
```

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)
```
MySQL [coms]> explain for connection 2281598777821235421;
+------------------------------+---------+----------+-----------+---------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------+------+
| id                           | estRows | actRows  | task      | access object       | execution info                                                                                                                                                                                                                                                                                                              | operator info                                                                                                                                                                                                                                                                                       | memory | disk |
+------------------------------+---------+----------+-----------+---------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------+------+
| Projection_4                 | 0.00    | 0        | root      |                     | time:0s, loops:0                                                                                                                                                                                                                                                                                                            | coms.pk_snpre10114.col2                                                                                                                                                                                                                                                                             | N/A    | N/A  |
| └─Selection_8                | 0.00    | 0        | root      |                     | time:0s, loops:0                                                                                                                                                                                                                                                                                                            | in(coms.pk_snpre10114.col1, "0x61219f79c90d3541f70e", 5501707547099269248, "0xec43efd30131dea2cb8b"), or(isnull(coms.pk_snpre10114.col2), in(coms.pk_snpre10114.col2, "呣丼蒢咿卻鹻铴础湜僂頃ǆ縍套衞陀碵碼幓9", "鹹楞睕堚尛鉌翡佾搁紟精廬姆燵藝潐楻翇慸嵊"))                                        | N/A    | N/A  |
|   └─TableReader_7            | 0.00    | 0        | root      |                     | time:0s, loops:0, cop_task: {num: 7990466, max: 223.8ms, min: 168.2µs, avg: 793.6µs, p95: 2.47ms, max_proc_keys: 2, p95_proc_keys: 0, tot_proc: 27.3s, tot_wait: 3m57.8s, rpc_num: 7992372, rpc_time: 1h43m57.7s, copr_cache_hit_ratio: 0.96, distsql_concurrency: 15}, backoff{regionMiss: 1.01s, regionScheduling: 6ms}   | data:Selection_6                                                                                                                                                                                                                                                                                    | N/A    | N/A  |
|     └─Selection_6            | 0.00    | 0        | cop[tikv] |                     | tikv_task:{proc max:4ms, min:0s, avg: 2.62ms, p80:3ms, p95:4ms, iters:7997090, tasks:7997090}, scan_detail: {total_process_keys: 661848, total_process_keys_size: 90673176, total_keys: 1654620, get_snapshot_time: 1m3.5s, rocksdb: {key_skipped_count: 661848, block: {cache_hit_count: 1985544}}}                        | or(isnull(coms.pk_snpre10114.col2), in(coms.pk_snpre10114.col2, "呣丼蒢咿卻鹻铴础湜僂頃ǆ縍套衞陀碵碼幓9", "鹹楞睕堚尛鉌翡佾搁紟精廬姆燵藝潐楻翇慸嵊"))                                                                                                                                              | N/A    | N/A  |
|       └─TableRangeScan_5     | 2.00    | 16002212 | cop[tikv] | table:PK_SNPRE10114 | tikv_task:{proc max:4ms, min:0s, avg: 2.62ms, p80:3ms, p95:4ms, iters:7998953, tasks:7998953}                                                                                                                                                                                                                               | range:[0x61219F79C90D3541F70E,0x61219F79C90D3541F70E], [0xEC43EFD30131DEA2CB8B,0xEC43EFD30131DEA2CB8B], [0x35353031373037353437303939323639323438,0x35353031373037353437303939323639323438], keep order:false, stats:pseudo                                                                         | N/A    | N/A  |
+------------------------------+---------+----------+-----------+---------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------+------+
5 rows in set (5.55 sec)
```
### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->
```
| Release Version: v6.2.0
Edition: Community
Git Commit Hash: a399181d861e21a1dbfc3ed00228866a1acd02c9
Git Branch: heads/refs/tags/v6.2.0
UTC Build Time: 2022-07-30 10:09:58
GoVersion: go1.18.2
Race Enabled: false
TiKV Min Version: 6.2.0-alpha
Check Table Before Drop: false
Store: tikv |
```

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: close #36876

Problem Summary:

### What is changed and how it works?

The key range is not sorted ... (and cause dead loop when coprocessor paging handle the remain key range)

### Check List

Tests 

- [X] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by XuHuaiyu — APPROVED** [review]

**Review by wshwsh12 — APPROVED** [review]

File: executor/distsql_test.go [hunk 1]
```diff
@@ -613,4 +613,19 @@ func TestCoprocessorPagingReqKeyRangeSorted(t *testing.T) {
...
+
+	tk.MustExec("CREATE TABLE `PK_SNPRE10114` (" +
+		"`COL1` varbinary(10) NOT NULL DEFAULT 'S'," +
+		"`COL2` varchar(20) DEFAULT NULL," +
+		"`COL4` datetime DEFAULT NULL," +
+		"`COL3` bigint(20) DEFAULT NULL," +
+		"`COL5` float DEFAULT NULL," +
+		"PRIMARY KEY (`COL1`) CLUSTERED)")
+	tk.MustExec(`prepare stmt from 'SELECT * FROM PK_SNPRE10114 WHERE col1 IN (?, ?, ?) AND (col2 IS NULL OR col2 IN (?, ?)) AND (col3 IS NULL OR col4 IS NULL);';`)
+	tk.MustExec(`set @a=0x0D5BDAEB79074756F203, @b=NULL, @c=0x6A911AAAC728F1ED3B4F, @d="鏖秿垙麜濇凗辯Ũ卮伄幖轒ƀ漭蝏雓轊恿磔徵", @e="訇廵纹髺釖寒近槩靏詗膦潳陒錃粓悧闒摔)乀";`)
```
Comment (ngaut) [inline executor/distsql_test.go:625]: What do these magic number/string mean?

Comment (tiancaiamao) [reply executor/distsql_test.go:625]: It's found by our comp test and the case seems like randomly generated.

**Pull request merged**